File: android-sdk-ui/src/com/appboy/ui/inappmessage/IInAppMessageAnimationFactory.java
package com.appboy.ui.inappmessage;

import android.view.animation.Animation;

import com.appboy.models.IInAppMessage;

public interface IInAppMessageAnimationFactory {

  /**
   * This method returns the animation that will be used to animate the message as it enters the screen.
   * @return animation that will be applied to the in-app message view using
   * {@link android.view.View#setAnimation}
   */
  Animation getOpeningAnimation(IInAppMessage inAppMessage);

  /**
   * This method returns the animation that will be used to animate the message as it exits the screen.
   * @return animation that will be applied to the in-app message view using
   * {@link android.view.View#setAnimation}
   */
  Animation getClosingAnimation(IInAppMessage inAppMessage);
}

File: android-sdk-ui/src/main/java/com/braze/ui/inappmessage/IInAppMessageAnimationFactory.kt
package com.braze.ui.inappmessage

import android.view.animation.Animation
import com.braze.models.inappmessage.IInAppMessage

interface IInAppMessageAnimationFactory {
    /**
     * This method returns the animation that will be used to animate the message as it enters the screen.
     * @return animation that will be applied to the in-app message view using
     * [android.view.View.setAnimation]
     */
    fun getOpeningAnimation(inAppMessage: IInAppMessage): Animation?

    /**
     * This method returns the animation that will be used to animate the message as it exits the screen.
     * @return animation that will be applied to the in-app message view using
     * [android.view.View.setAnimation]
     */
    fun getClosingAnimation(inAppMessage: IInAppMessage): Animation?
}
